galleries/examples/images_contours_and_fields/image_antialiasing.py

```diff
@@ -45,71 +40,213 @@
+# *interpolation_stage* of 'rgba'/'antialiased' for most down-sampling
+# situations (last panel).
+
+a = alarge + 1
+cmap = plt.get_cmap('RdBu_r')
+cmap.set_under('yellow')
+cmap.set_over('limegreen')
+
+fig, axs = plt.subplots(1, 3, figsize=(7, 3), layout='constrained')
+for ax, interp, space in zip(axs.flat,
+                             ['hanning', 'nearest', 'hanning', ],
+                             ['data', 'data', 'rgba']):
+    im = ax.imshow(a, interpolation=interp, interpolation_stage=space,
+                   cmap=cmap, vmin=0, vmax=2)
+    title = f"interpolation='{interp}'\nstage='{space}'"
+    if ax == axs[2]:
+        title += '\nDefault'
+    ax.set_title(title, fontsize='medium')
+fig.colorbar(im, ax=axs, extend='both', shrink=0.8)
+
+# %%
+# Up-sampling
+# ===========
+#
+# If we upsample, then we can represent a data pixel by many image or screen pixels.
+# In the following example, we greatly over-sample the small data matrix.
+
+np.random.seed(19680801+9)
+a = np.random.rand(4, 4)
+
+fig, axs = plt.subplots(1, 2, figsize=(6.5, 4), layout='compressed')
+axs[0].imshow(asmall, cmap='viridis')
+axs[0].set_title("interpolation='antialiased'\nstage='antialiased'")
+axs[1].imshow(asmall, cmap='viridis', interpolation="nearest",
+              interpolation_stage="data")
+axs[1].set_title("interpolation='nearest'\nstage='data'")
 plt.show()
 
+# %%
+# The *interpolation* keyword argument can be used to smooth the pixels if desired.
+# However, that almost always is better done in data space, rather than in RGBA space
+# where the filters can cause colors that are not in the colormap to be the result of
+# the interpolation.  In the following example, note that when the interpolation is
+# 'rgba' there are red colors as interpolation artifacts.  Therefore, the default
+# 'antialiased' choice for *interpolation_stage* is set to be the same as 'data'
+# when up-sampling is greater than a factor of three:
+
+fig, axs = plt.subplots(1, 2, figsize=(6.5, 4), layout='compressed')
+im = axs[0].imshow(a, cmap='viridis', interpolation='sinc', interpolation_stage='data')
+axs[0].set_title("interpolation='sinc'\nstage='data'\n(default for upsampling)")
+axs[1].imshow(a, cmap='viridis', interpolation='sinc', interpolation_stage='rgba')
+axs[1].set_title("interpolation='sinc'\nstage='rgba'")
+fig.colorbar(im, ax=axs, shrink=0.7, extend='both')
+
+# %%
+# Avoiding resampling
+# ===================
+#
+# It is possible to avoid resampling data when making an image.  One method is
+# to simply save to a vector backend (pdf, eps, svg) and use
+# ``interpolation='none'``.  Vector backends allow embedded images, however be
+# aware that some vector image viewers may smooth image pixels.
+#
+# The second method is to exactly match the size of your axes to the size of
+# your data. in the following, the figure is exactly 2 inches by 2 inches, and
+# the dpi is 200, so the 400x400 data is not resampled at all. If you download
```

Comment:
I think there is some normal/high dpi magic going on with saved images so it is either 100 or 200 based on the screen the reader has.

Reply:
OK< I put the dpi in as hard coded.  Not sure what happens for a low dpi screen if I do this.